File: go/53/53.clj
(def factorial 
  (fn  [n]
    (loop [cnt n acc 1]
      (if (zero? cnt) acc
        (recur (dec cnt) (* acc cnt))))))

(defn fact [n]
  (memoize (factorial n)))

(defn thing [n r]
  (/ (factorial n) (* (factorial r) (factorial (- n r)))))

(defn thing-m [n r]
  (memoize (thing n r)))

(defn apply-thing [n] 
  (println n)
  (pmap (fn [r] (thing n r)) (range 1 n)))

(defn bigdaddy? [n]
  (> n 1000000))

;(filter bigdaddy? (set (pmap apply-thing (range 1 1000))))
(filter bigdaddy? (seq (map apply-thing (range 1 100))))

File: go/53/53.py
def fact(x):
 if x == 0:
    return 1
 return x * fact(x - 1)


def thing( n, r ):
  return fact(n)/(fact(r)*fact(n-r))

sum = 0
for n in range(101):
  print n
  for r in range(1,n+1):
    if (thing(n,r) > 1000000):
      sum += 1

print sum
